bot.py

```
@@ -140,63 +163,107 @@ def pr_url(self):
         )
 
-    def message_if_unchecked(self):
+    async def message_if_unchecked(self):
         """
         Send a message next morning if any boxes are not yet checked off
         """
         unchecked_authors = get_unchecked_authors(self.org, self.repo, self.version)
         if unchecked_authors:
             slack_usernames = self.translate_slack_usernames(unchecked_authors)
-            self.say(
+            await self.say(
                 "What an unexpected surprise! "
                 "The following authors have not yet checked off their boxes: {}".format(
                     ", ".join(slack_usernames)
                 )
             )
 
+    async def delay_message(self):
+        """sleep until 10am next day, then message"""
+        now = datetime.now()
+        tomorrow_at_10 = next_workday_at_10(now)
+        await asyncio.sleep((tomorrow_at_10 - now).total_seconds())
+        await self.message_if_unchecked()
+
+    async def handle_message(self, content, loop):
+        """handle the message"""
+        if word_in('release', content):
+            version = get_version_number(content)
+            if not version:
+                await self.say('Could not find version number')
+            else:
+                self.set_repo_details(version)
+                if 'start release' in content:
+                    loop.create_task(self.delay_message())
+                    await self.do_release()
+                    # await bot.wait_for_checkboxes()
+                elif 'finish release' in content:
+                    await self.finish_release()
+
+
+def get_version_number(message):
+    """return version number at the end of the message"""
+    pattern = re.compile('release [0-9.]+$')
+    match_obj = pattern.search(message)
+    if match_obj:
+        pieces = message.split()
+        pattern = re.compile('^[0-9.]+$')
+        return [text for text in pieces if pattern.match(text)][0]
+    else:
+        return ""
```

Can we raise an exception here instead? If the version number is not formatted correctly it should be treated as an error case, then the exception should be handled appropriately in the calling function